### PR TITLE
Upgrade Gradle Wrapper to 8.4

### PR DESCRIPTION
**Pull request by tisonkun**

### CODE REVIEW

File: gradlew.bat
Comment (tisonkun) [inline gradlew.bat:1]: Seem invisible characters change. I don't know how to configure it preventing unnecessary diff.

The command I ran is `gradle wrapper --gradle-version 8.4 ` with Gradle 8.4.

File: gradlew [hunk 1]
```diff
@@ -1,8 +1,19 @@
 #!/bin/sh
 
 #
-# Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
-# Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+# Copyright © 2015-2021 the original authors.
```
Comment (tisonkun) [inline gradlew:4]: Of course this file is copied from the original author.

File: gradlew.bat
Comment (ice1000) [reply gradlew.bat:1]: I don't mind

File: gradlew (hunk 1 above)
Comment (ice1000) [reply gradlew:4]: Sure

**Comment by ice1000** [comment]
CI fails

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,8 +1,7 @@
-# Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
-# Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
```
Comment (ice1000) [inline gradle/wrapper/gradle-wrapper.properties:2]: Shall not delete this

File: gradlew.bat
Comment (ice1000) [reply gradlew.bat:1]: Wait I do. This is very bad. Is it caused by line ending? Are you using crlf?

**Review by ice1000 — CHANGES_REQUESTED** [review]
Main reason to request changes is about CI failure

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (tisonkun) [reply gradle/wrapper/gradle-wrapper.properties:2]: I can add it back, while this file is a generated file and generally we can place license header for such file - I'm curious why you think it's necessary or just preference.

File: gradlew.bat
Comment (tisonkun) [reply gradlew.bat:1]: > Is it caused by line ending

I think so. This is the result of `gradle wrapper --gradle-version 8.4` on macOS. Did you manually change the line separator after upgrade gradle wrapper before? Or there is some settings I miss to set up local dev env.

**Comment by tisonkun** [comment]
> CI fails

The direct failure is fixed. You may retrigger the workflows :D

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (ice1000) [reply gradle/wrapper/gradle-wrapper.properties:2]: The license is also generated

File: gradlew.bat
Comment (ice1000) [reply gradlew.bat:1]: I configured git to do that.

**Comment by ice1000** [comment]
I'll be doing this myself. I don't like your line separator changes, I think those are unnecessary.

**Comment by tisonkun** [comment]
> your line separator changes

I configure `git config core.autocrlf true` and regenerate the wrapper script. May you take another look.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (tisonkun) [reply gradle/wrapper/gradle-wrapper.properties:2]: Added back.

**Comment by ice1000** [comment]
Perfect.

bors r+

**Review by ice1000 — APPROVED** [review]
